lib/model.js

```diff
@@ -411,6 +413,61 @@ assign(derived, {
   }
 });
 
+/**
+ * Adds auth info to URL. The connection model builds two URLs.
+ * driverUrl - for the driver with the password included.
+ * safeUrl - for the UI with stars instead of password.
+ *
+ * @param {Object} options - Has only isPasswordProtected propery.
+ *
+ * @returns {String} - URL with auth.
+ */
+function addAuthToUrl({ isPasswordProtected }) {
+  let username = '';
+  let password = '';
+  let authField = '';
+  let result = this.baseUrl;
+
+  // Post url.format() workaround for
+  // https://github.com/nodejs/node/issues/1802
+  if (
+    this.authStrategy === 'MONGODB' ||
+    this.authStrategy === 'SCRAM-SHA-256'
+  ) {
+    username = encodeURIComponent(this.mongodbUsername);
+    password = isPasswordProtected ? '*****' : encodeURIComponent(this.mongodbPassword);
+    authField = format('%s:%s', username, password);
+  } else if (this.authStrategy === 'LDAP') {
+    username = encodeURIComponent(this.ldapUsername);
+    password = isPasswordProtected ? '*****' : encodeURIComponent(this.ldapPassword);
+    authField = format('%s:%s', username, password);
+  } else if (this.authStrategy === 'X509') {
+    username = encodeURIComponent(this.x509Username);
+    authField = username;
+  } else if (this.authStrategy === 'KERBEROS' && this.kerberosPassword) {
+    username = encodeURIComponent(this.kerberosPrincipal);
+    password = isPasswordProtected ? '*****' : encodeURIComponent(this.kerberosPassword);
+    authField = format('%s:%s', username, password);
+  } else if (this.authStrategy === 'KERBEROS') {
+    username = encodeURIComponent(this.kerberosPrincipal);
+    authField = format('%s:', username);
+  }
+
+  // The auth component comes straight after `the mongodb://`
+  // so a single string replace should always work
+  result = result.replace('AUTH_TOKEN', authField, 1);
+
+  if (['LDAP', 'KERBEROS'].includes(this.authStrategy)) {
+    result = `${result}&authSource=$external`;
+  }
+
+  if (this.kerberosCanonicalizeHostname) {
```

Comment:
I don't think there's a chance that someone would put `kerberosCanonicalizeHostname` as true without having kerberos, but the behavior would change here since before it was wrapped in a kerberos check. I don't think we need to worry about it though?

Reply:
Hmm thinking about it more maybe we should keep it in a `kerberos` check just to be safe. I'm sure there's someone out there copy pasting url properties without checking xD

Reply:
Wrapped it back in Kerberos check.